test/system/150-login.bats

```diff
@@ -398,6 +398,84 @@ EOF
     assert "$output" =~ "--retry-delay .*pull failures \(default \"5s\"\)"
 }
 
+@test "push and pull zstd chunked image" {
+    iid=localhost:${PODMAN_LOGIN_REGISTRY_PORT}/$(random_string 10 | tr A-Z a-z)
+
+    run_podman push \
+               --compression-format zstd:chunked \
+               --cert-dir ${PODMAN_LOGIN_WORKDIR}/trusted-registry-cert-dir \
+               --creds ${PODMAN_LOGIN_USER}:${PODMAN_LOGIN_PASS} \
+               $IMAGE $iid
+
+
+    image_archive=$PODMAN_TMPDIR/myimage-$(random_string 8).tar
+
+    run_podman save $IMAGE -o $image_archive
+    run_podman rmi $IMAGE
+
+    run_podman pull \
+               --cert-dir ${PODMAN_LOGIN_WORKDIR}/trusted-registry-cert-dir \
+               --creds ${PODMAN_LOGIN_USER}:${PODMAN_LOGIN_PASS} \
+               $iid
+
+    run_podman load -i $image_archive
+
+    run_podman diff $IMAGE $iid
+    assert "$output" "aa"
+
+    dockerfile2=$PODMAN_TMPDIR/Dockerfile.2
+    cat >$dockerfile2 <<EOF
+FROM $iid
+RUN touch /new-file
+EOF
+
+    dockerfile3=$PODMAN_TMPDIR/Dockerfile.3
+    cat >$dockerfile3 <<EOF
+FROM $iid
+RUN touch /new-file-2
+EOF
+
+    iid2=localhost:${PODMAN_LOGIN_REGISTRY_PORT}/$(random_string 10 | tr A-Z a-z)
+    iid3=localhost:${PODMAN_LOGIN_REGISTRY_PORT}/$(random_string 10 | tr A-Z a-z)
+
+    run_podman build --squash-all -t $iid2 -f $dockerfile2 .
+    run_podman build --squash-all -t $iid3 -f $dockerfile3 .
+
+    run_podman push \
+               --compression-format zstd:chunked \
+               --cert-dir ${PODMAN_LOGIN_WORKDIR}/trusted-registry-cert-dir \
+               --creds ${PODMAN_LOGIN_USER}:${PODMAN_LOGIN_PASS} \
+               $iid2
+
+    run_podman push \
+               --compression-format zstd:chunked \
+               --cert-dir ${PODMAN_LOGIN_WORKDIR}/trusted-registry-cert-dir \
+               --creds ${PODMAN_LOGIN_USER}:${PODMAN_LOGIN_PASS} \
+               $iid3
+
+    run_podman diff $iid3 $iid2
+    sorted_output=$(echo $output | sort)
```

Comment:
I didn't expect this to work, and it doesn't, and for many reasons. First and most important, `echo` is weird in ways that one has no choice but to shrug and accept:
```
    echo  $multilinestring    <<<---- collapses to just one line!
    echo "$multilinestring"   <<<---- with quotes, it preserves newlines
```
so anyhow, the `sort` there is a NOP. I'm not sure what it is you're trying to check, and it's late in my day, so I'm sorry not to be able to offer a suggestion for how to solve this.